Sources/Commands/SwiftPackageTool.swift

```diff
@@ -966,3 +1051,207 @@ fileprivate func logPackageChanges(changes: [(PackageReference, Workspace.Packag
     }
     stream.flush()
 }
+
+internal struct PackageTemplate {
+    struct Directories {
+        let sources: RelativePath
+        let tests: RelativePath?
+        let createSubDirectoryForModule: Bool
+    }
+    
+    enum PackageType: String {
+        case executable
+        case library
+        case systemModule = "system-module"
+        case empty
+        case manifest
+    }
+    
+    let directories: Directories
+    let type: PackageType
+}
+
+extension InitPackage.PackageTemplate {
+    internal init(template: PackageTemplate) throws {
+        
+        let packageType: InitPackage.PackageType
+        switch template.type {
+        case .executable:
+            packageType = .executable
+        case .library:
+            packageType = .library
+        case .systemModule:
+            packageType = .systemModule
+        case .empty:
+            packageType = .empty
+        case .manifest:
+            packageType = .manifest
+        }
+        
+        self.init(sourcesDirectory: template.directories.sources,
+                  testsDirectory: template.directories.tests,
+                  createSubDirectoryForModule: template.directories.createSubDirectoryForModule,
+                  packageType: packageType)
+    }
+}
+
+internal var useNewBehaviour: Mode {
+    get {
+        switch (ProcessEnv.vars["SWIFTPM_ENABLE_PACKAGE_CREATE"].map { $0.lowercased() }) {
+        case "true":
+            return .new
+        default:
+            return .legacy
+        }
+    }
+}
+
+internal enum Mode {
+    case new
+    case legacy
+}
+
+internal func getSwiftPMDefaultTemplate(type: InitPackage.PackageType,
+                                        sources: RelativePath = .init("./Sources"),
+                                        tests: RelativePath? = nil,
+                                        createSubDirectoryForModule: Bool = false) -> PackageTemplate {
```

Comment:
this can now return InitPackage.PackageTemplate since we dont need the other type now that we are not parsing JSON